virtualization-and-containers/material/vagrant_slides.md

```diff
@@ -0,0 +1,197 @@
+
+---
+
+## What for?
+
+- Developing software
+  - Set up consistent development environment
+  - Share environment with others
+- Testing software, workflows...
+  - Disposable environments
+  - Consistent workflows
+- More general
+  - Easy way to setup virtual machine
+
+---
+
+## Advantages
+
+- Strong focus on workflow consistency
+- Avoid usage of (VirtualBox) CLI
+- Store in Git-friendly format
+- For us:
+  - Management of VirtualBox VMs (testing, developing...)
+  - Sharing of VMs (debugging, workshops...)
+
+---
+
+## Building Block
+
+- Provisioners
+  - Tool for configuring the boxes (install software on machine)
+  - Examples: shell scripts, Chef, Puppet..
+- Providers
+  - "Backend" the box is based on
+  - Examples: VirtualBox, Hyper-V, AWS, Docker...
+
+---
+
+## Structure of Vagrant Box
+
+- File `Vagrantfile`
+  - Contains configuration of VM
+  - Ruby script
+- Supplementary files, scripts etc. for configuring VM
+  - Example: `bootstrap.sh`
+- Files are usually text files (Git friendly)
+
+---
+
+## Setting up a Virtual Machine
+
+- Install VirtualBox
+- Install Vagrant (ideallt from homepage)
+- Create `Vagrantfile` and specify image
+
+  ```ruby
+  Vagrant.configure("2") do |config|
+    config.vm.box = "ajaust/sse-first-steps"
+    config.vm.box_version = "0.1.0"
+  end
+  ```
+
+- Alternative:
+
+  ```
+  vagrant init ajaust/sse-first-steps
+  ```
+
+- Start VM with `vagrant up`
+
+---
+
+## Some Useful Commands 1/2
+
+- `vagrant init USERNAME/BOXNAME or URL`
+  - Initializes repository in current directory
+  - Creates `Vagrantfile` -> Text file (Git friendly)
+- `vagrant up`
+  - Create VM and start it
+  - VM is stored in `${HOME}/.vagrant.d/boxes` (overruled by `VAGRANT_HOME`)
+- `vagrant ssh`
+  - Connect to VM via ssh
+- `vagrant destroy`
+  - Stops the virtual machine and removes its disk
+
+---
+
+## Some Useful Commands 2/2
+
+- `vagrant box list`
+  - Show available boxes
+- `vagrant box remove NAMEOFBOX`
+  - Remove a box completely
+- `vagrant suspend`
+  - Suspends machine (sleep)
+- `vagrant halt`
+  - Shuts down machine keeping changes
+- `vagrant reload --provision` or `vagrant provision`
+  - Rebuild (partially) build image
+
+---
+
+## Exchanging Files
+
+- Vagrant mounts shared folders
+  - Check output when VM starts
+
+  ```bash
+  default: Mounting shared folders...
+  ```
+
+  - File can be changed from Guest or Host
+- Default:
+  - Directory containing `Vagrantfile` mounted to `/vagrant` in the Guest
+
+---
+
+## Vagrant Cloud
+
+- Repository of premade boxes: [https://app.vagrantup.com/](https://app.vagrantup.com/)
+- Pack box:
+
+  ```
+  vagrant package --base "NAMEOFVM" --output BOXNAME.box
+  ```
+
+  - `NAMEOFVM` is name as shown in VirtualBox
+  - `BOXNAME` name to store box to
+- [Publish own boxes](https://www.vagrantup.com/docs/providers/virtualbox/boxes) by uploading to [Vagrant Cloud](https://app.vagrantup.com/)
+
+---
+
+## Demo
```

Comment:
I have the feeling the demo comes too late. Or do you already show some live things earlier when you show "some useful commands"?

Reply:
You are right about that. I had already made a note about that and moved one demo section just after the "useful commands" part.